crates/ty_python_semantic/src/types.rs

```diff
@@ -543,8 +545,10 @@ pub enum Type<'db> {
     LiteralString,
     /// A bytes literal
     BytesLiteral(BytesLiteralType<'db>),
-    /// A heterogeneous tuple type, with elements of the given types in source order.
-    // TODO: Support variable length homogeneous tuple type like `tuple[int, ...]`.
+    /// An instance of the builtin `tuple` class.
+    /// TODO: Consider removing this in favor of `NominalInstance`. This is currently stored as a
+    /// separate variant partly for historical reasons, and partly to allow us to easily
+    /// distinguish tuples since they occur so often.
```

Comment:
The newest changes on this PR now technically make the `Type::Tuple` variant redundant, since we can represent that as a `NominalInstance` of an appropriate `tuple` generic alias. I experimented with actually removing that, but there are a _lot_ of places where we have special-case logic that would have to move into `NominalInstance`, and it also seemed to introduced a bad performance regression. So I'm leaving the dedicated `Type::Tuple` variant in place for now. We can investigate removing it in a future PR.